crates/kornia-core/src/storage.rs

```diff
@@ -313,4 +343,21 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_tensor_storage_into_vec() {
+        let allocator = CpuAllocator;
+        let original_vec = vec![1, 2, 3, 4, 5];
+        let original_vec_ptr = original_vec.as_ptr();
+        let original_vec_capacity = original_vec.capacity();
+
+        let storage = TensorStorage::<i32, _>::from_vec(original_vec, allocator);
+
+        // Convert the storage back to a vector
+        let result_vec = storage.into_vec();
+
+        // check NO copy
+        assert_eq!(result_vec.capacity(), original_vec_capacity);
+        assert!(std::ptr::eq(result_vec.as_ptr(), original_vec_ptr));
```

Comment:
@decahedron1 this test seems to fail, which means that we enter in the Err(buf) case

Reply:
maybe there's something wrong with the `TensorStorage::from_vec` which underneath uses [`Buffer::from_custom_allocation`](https://docs.rs/arrow/latest/arrow/buffer/struct.Buffer.html#method.from_custom_allocation)

Reply:
UPDATE: i think i fixed ... but I'm missing something around how to tie with the provided allocator. My idea here was that the user can provider custom allocators. eg. CudaAllocator to hold CudaImages to interop easily with other frameworks like ort, vpi, etc  to not move memory forth and back

Reply:
Hmm, what was causing the test failure? The fix in https://github.com/kornia/kornia-rs/pull/125/commits/1642803ab4c8113b772b0ef8c839cfe3d02ae622 doesn't seem obvious to me, but again, I know nothing about `arrow_buffer` 🤔

Integration with CUDA buffers would have to go through a different API, [`TensorRefMut::from_raw`](https://docs.rs/ort/2.0.0-rc.5/ort/type.TensorRefMut.html#method.from_raw). It just needs a pointer and shape, so Kornia's existing APIs suffice. `TensorRefMut`s can be used zero-copy in session inputs just like a `Value`, but since the memory is user-managed, it does have a lifetime bound. See ort's `cudarc` example: https://github.com/pykeio/ort/blob/60f6ecae9064a75a474678998af42ea87067b1b1/examples/cudarc/src/main.rs#L35-L43

Reply:
> Hmm, what was causing the test failure?
I might need to investigate a bit more the `Buffer::from_custom_allocation`, maybe is doing copy because later the test checking ptrs is failing.

> Integration with CUDA buffers would have to go through a different API, 
I'll play with it , thanks for the pointers.